### PR TITLE
docker: add core18 snap that snapcraft now uses as a base

### PR DESCRIPTION
**Pull request by sergiusens**
Also update comments for correctness and syntax.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: docker/candidate.Dockerfile [hunk 1]
```diff
@@ -8,17 +8,26 @@ RUN apt-get install --yes \
...
 RUN unsquashfs -d /snap/core/current core.snap
 
-# Grab the snapcraft snap from the candidate channel and unpack it in the proper place
+# Grab the core18 snap (which snapcraft uses as a base) from the stable channel
+# and unpack it in the proper place.
+RUN curl -L $(curl -H 'X-Ubuntu-Series: 16' 'https://api.snapcraft.io/api/v1/snaps/details/core18' | jq '.download_url' -r) --output core18.snap
+RUN mkdir -p /snap/core18
+RUN unsquashfs -d /snap/core18/current core18.snap
+
+# Grab the snapcraft snap from the beta channel and unpack it in the proper
```
Comment (rajivshah3) [inline docker/candidate.Dockerfile]:
```suggestion
# Grab the snapcraft snap from the candidate channel and unpack it in the proper
```

File: docker/edge.Dockerfile [hunk 2]
```diff
@@ -8,17 +8,26 @@ RUN apt-get install --yes \
...
 RUN unsquashfs -d /snap/core/current core.snap
 
-# Grab the snapcraft snap from the edge channel and unpack it in the proper place
+# Grab the core18 snap (which snapcraft uses as a base) from the stable channel
+# and unpack it in the proper place
+RUN curl -L $(curl -H 'X-Ubuntu-Series: 16' 'https://api.snapcraft.io/api/v1/snaps/details/core18' | jq '.download_url' -r) --output core18.snap
+RUN mkdir -p /snap/core18
+RUN unsquashfs -d /snap/core18/current core18.snap
+
+# Grab the snapcraft snap from the beta channel and unpack it in the proper
```
Comment (rajivshah3) [inline docker/edge.Dockerfile]:
```suggestion
# Grab the snapcraft snap from the edge channel and unpack it in the proper
```

File: docker/stable.Dockerfile [hunk 3]
```diff
@@ -8,17 +8,26 @@ RUN apt-get install --yes \
...
 RUN unsquashfs -d /snap/core/current core.snap
 
-# Grab the snapcraft snap from the stable channel and unpack it in the proper place
+# Grab the core18 snap (which snapcraft uses as a base) from the stable channel
+# and unpack it in the proper place.
+RUN curl -L $(curl -H 'X-Ubuntu-Series: 16' 'https://api.snapcraft.io/api/v1/snaps/details/core18' | jq '.download_url' -r) --output core18.snap
+RUN mkdir -p /snap/core18
+RUN unsquashfs -d /snap/core18/current core18.snap
+
+# Grab the snapcraft snap from the beta channel and unpack it in the proper
```
Comment (rajivshah3) [inline docker/stable.Dockerfile]:
```suggestion
# Grab the snapcraft snap from the stable channel and unpack it in the proper
```

File: docker/candidate.Dockerfile (hunk 1 above)
Comment (sergiusens) [reply docker/candidate.Dockerfile]: thanks!

**Comment by abitrolly** [comment]
That's great! The only thing that upsets me is the different authorship from the commit that is signed by me. :D Did you use `git cherry-pick` for merging  #2851?

**Pull request merged**